### ISSUE TITLE
`UnnecessaryBracesAroundTrailingLambda` false positive on vararg

### ISSUE THREAD

**Issue by sulume** [detekt/detekt#6593]
## Expected Behavior

It should pass the case where the last function parameter on call site (that is not the last one by definition) cannot be a trailing lambda (placed outside the parentheses).

## Observed Behavior

Reported `"Braces around trailing lambda can be removed."`

## Steps to Reproduce

```kotlin
fun takeLambdaFollowedByVararg(f: () -> Unit, vararg x: Unit) = f()

// This is detected noncompliant with UnnecessaryBracesAroundTrailingLambda.
takeLambdaFollowedByVararg({})
// Failed to compile if removed braces.
takeLambdaFollowedByVararg {}
```

## Context

A real world example you could pass a lambda as the last parameter on call site but not replaceable with a trailing lambda:
https://github.com/micrometer-metrics/micrometer/blob/2e0fa1b2dbcd1b93402fe99141c125ba417713ee/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/grpc/MetricCollectingServerInterceptor.java#L85

## Your Environment

* Version of detekt used: 1.23.1
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: Ubuntu 22.04.3
* Link to your project (if it's a public repository):

**Comment by atulgpt** [comment]
Hi, @sulume I have added the TC for this https://github.com/detekt/detekt/pull/6596 and it is working fine the test. Is this happening only with certain Kotlin version or even for the latest version?

**Comment by sulume** [comment]
Thank you for your quick response and for looking into this.

Sorry that my report wasn't sufficient to reproduce the issue. What I thought about the cause and reported seems missing the point.

The issue is in a private repository and I'm still trying to extract and make an example from it (because of a complex build setup) so I don't have correct steps now. I'm not sure if this is actually a bug or some other kind, so please let me know if I should reopen or open another when I could make a sufficient report.

I'm wondering which parts of my environment are related but trying to give some more info anyway.

I build it with bazel 6.3.0, [rules_kotlin 1.8](https://github.com/bazelbuild/rules_kotlin/tree/v1.8), kotlin 1.8.21, and then run detekt-cli-1.23.1 with jre zulu-17 on local mac and on ubuntu with github actions (both work almost the same).

The problematic expression of actual code is following, depending on [micrometer-core 1.10.8](https://github.com/micrometer-metrics/micrometer/tree/v1.10.8) via [maven](https://repo1.maven.org/maven2/io/micrometer/micrometer-core/1.10.8/):
```kotlin
MetricCollectingServerInterceptor(
    meterRegistry,
    UnaryOperator.identity(),
    { timer -> timer.publishPercentileHistogram() },
)
```

and the command to run detekt-cli is following:
```sh
curl -sSLO https://github.com/detekt/detekt/releases/download/v1.23.1/detekt-cli-1.23.1.zip
unzip detekt-cli-1.23.1.zip
# here several steps to build things for type resolution.
./detekt-cli-1.23.1/bin/detekt-cli \
  --config detekt-config.yml \
  --build-upon-default-config \
  --input java/ \
  --classpath $(find -L bazel-bin \
    -name '*.jar' \
    -not -path '*.runfiles/*' \
    -not -name '*-src.jar' \
    -not -name '*-gensrc.jar' \
    -not -name '*_deploy.jar' \
    -printf "%p;") \
  --report xml:detekt_report.xml
```

I'd be grateful for any help.

**Comment by atulgpt** [comment]
If you can reproduce the bug with the smaller reproducible example and then try to use that example with gradle setup then that will help in figuring out the root cause. Also, currently in the rule `UnnecessaryBracesAroundTrailingLambda` I am simply checking the Kotlin warning and surfacing that warning in the detekt. So this rule itself doesn't have any logic to it.

**Comment by BraisGabin** [comment]
Closing as there is no way to reproduce it. Don't hesitate to open a new issue or comment here if you can provide a reproducer.
